src/withDeprecatedProps.tsx

```diff
@@ -69,13 +81,14 @@ function withDeprecatedProps(WrappedComponent, componentName, deprecatedProps) {
         .reduce(this.transformProps, {});
 
       return (
-        <WrappedComponent {...transformedProps}>
-          {this.props.children || children /* eslint-disable-line react/prop-types */}
+        <WrappedComponent {...transformedProps as T}>
+          {this.props.children || children}
         </WrappedComponent>
       );
     }
   }
 
+  // @ts-ignore
```

Comment:
[curious] why does this need to be ts-ignore'd

Reply:
TypeScript complained that `displayName` property does not exist in the component, so you can't assign it.

I force-pushed, sou you can't see anymore, but `ts-ignore` complained about line
```js
WithDeprecatedProps.displayName = `withDeprecatedProps(${componentName})`;
```